front/app/modules/commercial/insights/admin/containers/Insights/Edit/InputsTable/index.tsx

```diff
@@ -47,44 +50,73 @@ const StyledTable = styled(Table)`
   }
 `;
 
+const StyledPagination = styled(Pagination)`
+  margin-top: 12px;
+`;
+
 const InputsTable = ({
+  location,
   params: { viewId },
   intl: { formatMessage },
 }: WithRouterProps & InjectedIntlProps) => {
-  const inputs = useInsightsInputs(viewId);
-  if (isNilOrError(inputs)) {
-    return null;
-  }
+  const options = {
+    ...(location?.query?.pageNumber
+      ? { pageNumber: parseInt(location?.query?.pageNumber, 10) }
+      : {}),
+  };
+
+  const { list: inputs, currentPage, lastPage } = useInsightsInputs(
+    viewId,
+    options
+  );
+
+  const handlePaginationClick = (newPageNumber) => {
+    clHistory.push({
```

Comment:
Should we do replace instead, here? Not sure how this is done elsewhere in the app but maybe the back button should lead to previous screen, rather than prev page.